relay-server/src/services/processor.rs

```diff
@@ -2645,6 +2649,43 @@ impl UpstreamRequest for SendMetricsRequest {
     }
 }
 
+/// Container for global and project level [`Quota`].
+#[derive(Copy, Clone)]
+pub struct DynamicQuotas<'a> {
```

Comment:
Wouldn't make this pub, it's just a helper for the processor.

Also the reason I thought we want to change the name to something more descriptive, but the argument that `Dynamic` is a convention for this sort of thing is good 👍 .

Reply:
how can the type be private and its constructor not ? 

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=537ebc15e46838757d1bb0295f59f73a#:~:text=Permalink%20to%20the%20playground

see comp error

Reply:
It's not used outside of this module (and that's what we want to prevent): https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=9c751fa63c54def6ced6aeb184c6235b

Reply:
Constructor does not have to be public for this either, but I like still to keep the visibility modifier as a way of signalling "this function is intended to be called from outside the struct" vs "this method is an implementation detail and probably shouldnt be called".

Reply:
alright, that makes sense to me, thanks!